### PR TITLE
finish chapter 4

### PR DESCRIPTION
**Pull request by 2H1EU8**
Finish all task in Chapter 4

### CODE REVIEW

File: p1_chapter4.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        array[i] = array[array.length - 1 -i]
+        array[array.length - 1 -i] = temp
+    }
+    return array
+}
+console.log(reversInPlace([1,2,3,4,5]))
+
+//A list
+
+const rayToList = (ray) => {
```
Comment (thanhbui27) [inline p1_chapter4.js]: rayToList ?? 
and param : ray ?

File: p1_chapter4.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    if (!array.length) {
+        return {};
```
Comment (thanhbui27) [inline p1_chapter4.js:42]: we can use return

File: p1_chapter4.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    let list = {
+        value: array[0]
+    };
+
+
+    if (array.length == 1) {
```
Comment (thanhbui27) [inline p1_chapter4.js]: u can use '===' instead '=='

File: p1_chapter4.js (hunk 2 above)
Comment (2H1EU8) [reply p1_chapter4.js:42]: this function is convert a array to object, so if array null we will return { }

**Pull request merged**